app/src/main/java/nil/nadph/qnotified/bridge/QQMessageFacade.java

```diff
@@ -37,14 +37,23 @@ public static void revokeMessage(Object msg) throws Exception {
         //if (istroop != 0) throw new IllegalArgumentException("istroop(" + istroop + ") is not supported");
         Object mgr = getMessageManager(istroop);
         try {
-            Object msg2 = invoke_static(DexKit.doFindClass(DexKit.C_MSG_REC_FAC), "a", msg, Initiator._MessageRecord(), Initiator._MessageRecord());
+            Object msg2 = invoke_static_any(DexKit.doFindClass(DexKit.C_MSG_REC_FAC), msg, Initiator._MessageRecord(), Initiator._MessageRecord());
             long t = (long) iget_object_or_null(msg2, "time");
             t -= 1 + 10f * Math.random();
             iput_object(msg2, "time", t);
-            String fuckingMethod = "a";
-            if (Utils.getHostVersionCode() >= QQVersion.QQ_8_4_8) fuckingMethod = "getMsgCache";
-            Object msgCache = invoke_virtual(getQQAppInterface(), fuckingMethod, DexKit.doFindClass(DexKit.C_MessageCache));
-            invoke_virtual(msgCache, "b", true, boolean.class, void.class);
+            Object msgCache = invoke_virtual_any(getQQAppInterface(), DexKit.doFindClass(DexKit.C_MessageCache));
+            String methodName = "b"; //Default method name for QQ
+            if (isTim(getApplication())) {
+                long version = getHostVersionCode();
```

Comment:
you may need to create a class like https://github.com/ferredoxin/QNotified/blob/master/app/src/main/java/me/singleneuron/util/QQVersion.java for tim. Do not use meaningless number anytime!
and put methods and so on in https://github.com/ferredoxin/QNotified/blob/master/app/src/main/java/me/singleneuron/qn_kernel/tlb/ConfigTable.kt